src/Microsoft.Diagnostics.Monitoring.WebApi/ActionContextExtensions.cs

```diff
@@ -14,68 +14,60 @@
 
 namespace Microsoft.Diagnostics.Monitoring.WebApi
 {
-    internal static class ActionContextExtensions
+    public sealed class ExecutionResult<T>
```

Comment:
nit: It may be better to make this immutable. Also, maybe add two static methods for `static ExecutionResult<T> Success<T>(T result)` and `static ExecutionResult<T> Failed<T>(ProblemDetails details)`